src/bindings/pygaia/scripts/classification/generate_classification_project.py

```diff
@@ -57,14 +66,41 @@ def generateProject(groundtruth_file, filelist_file, project_file, datasets_dir,
         print("track ids found in", groundtruth_file, "are inconsistent with", filelist_file)
         sys.exit(4)
 
+    print('Analizing the dataset to figure out which project template file to use...')
+
+    essentia_version = ''
+    if not template:
+        try:
+            versions = set([yaml.load(open(v))['metadata']['version']['essentia'] for v in fl.values()])
```

Comment:
it's a bit annoying that we can't do this in combination with another operation... that is, we're doing an open for all files just to get a single value, which for a large dataset might take a long time.
It might be good to use `.get('', {})` here so that any bad files don't cause the process to crash